File: backend/app/models/ml.py
from pydantic import BaseModel


class MLResultResponse(BaseModel):
    id: str
    model_type: str
    trained_at: str
    record_count: int
    accuracy_score: float | None = None
    predictions: list[dict] = []
    anomaly_record_ids: list[str] = []
    correlation_matrix: dict | None = None
    forecast_data: list[dict] | None = None
    job_id: str | None = None

File: frontend/src/types/ml.ts
export interface MLResult {
  id: string;
  model_type: string;
  trained_at: string;
  record_count: number;
  accuracy_score: number | null;
  predictions: Record<string, unknown>[];
  anomaly_record_ids: string[];
  correlation_matrix: Record<string, Record<string, number | null>> | null;
  forecast_data: Array<{ region: string; date: string; forecast_temp_c: number }> | null;
  job_id: string | null;
}